Styled Components + Jest Tests!

mdwagner opened the pull request:
**Highlights**

- Changed all React Native Styles to [Styled Components](https://www.styled-components.com). This change also allowed every styled component to have its' own folder structure.
- Added more [Jest](https://facebook.github.io/jest) tests 🎆. I didn't add a test for every single thing, but I did add at least 1 test per styled component. There is definitely room for improvement on testing!
- Upgraded all `React.PropTypes` to `PropTypes` [package](https://github.com/facebook/prop-types).
- Moved API keys into `env.json` file.
- Added [package](https://babeljs.io/docs/plugins/transform-inline-environment-variables) for reading environment variables.
- Added [package](https://github.com/tleunen/babel-plugin-module-resolver) for using aliases pointing to src folders.
- Added [Travis CI](https://travis-ci.org) config. [![Build Status](https://travis-ci.org/mdwagner/charlotte-lightrail.svg?branch=development)](https://travis-ci.org/mdwagner/charlotte-lightrail)
- Added [Codecov](https://codecov.io/) config. [![codecov](https://codecov.io/gh/mdwagner/charlotte-lightrail/branch/development/graph/badge.svg)](https://codecov.io/gh/mdwagner/charlotte-lightrail)
- Some more project refactoring, including using an `index.js` inside each folder for easier imports.
- Updated Read Me with more current information.

Shout out to @nickbreaton for help on Styled Components!
Please let me know if you have any questions!

brandonfancher commented:
Nice, @mdwagner. Thank you! Will review and test and provide feedback/merge.

brandonfancher commented on README.md:
```diff
@@ -28,21 +25,32 @@ If this is your first time running a React Native project on this machine, follo
 ### Keys/Secrets
 
-You will need to get your own Mapbox API Key. Globally find `[MAPBOX_API_KEY]` and replace with your key.
+You will need to get your own Mapbox API Key & Google Maps API Key. Place them in **`env.json`** before you start.
```
IIRC, the Google Maps API key stuff can be removed from the project. We're relying solely on Mapbox's Distance API. (We had started with Google's. It was better (included traffic in distance calculations), but we removed that because of the expenses and low API call limits and opted for Mapbox's instead. So removing the Google API stuff altogether would be an improvement.

mdwagner replied:
Yeah I can remove the Google stuff before it gets merged

brandonfancher replied:
Thanks!

brandonfancher commented:
Wow, this is a serious refactor! Nice!

mdwagner commented:
This is should be good to merge, if you don't have anything else to request. If you do, let me know.

brandonfancher commented:
Note for future reference: I was getting errors due to the react native packager looking in the wrong directory for assets. After researching this a bit, I learned that after messing with the `.babelrc` file, one must run `npm start -- --reset-cache` to get it to properly pick up on the `module-resolver` changes. After running that command, the app started up as expected. This probably wouldn't be a problem on a clean clone.

mdwagner commented:
@brandonfancher I can confirm that I have experienced that issue as well. I'm not sure if I want to keep the `module-resolver` package (because of that issue), but also because the maintainer doesn't use React Native and isn't looking to fix any of these issues. For now I'll keep it, but it might be removed in the future, depending on if it gives me any more trouble.

brandonfancher commented:
@mdwagner Yeah, your call. I'm fine keeping it (as long as we're aware that we have to run that command from time-to-time), but it does introduce one more complexity--compatibility with ESLint. If we keep it, we'd need to add the `eslint-import-resolver-babel-module` package (See https://github.com/tleunen/babel-plugin-module-resolver/tree/v2.7.0#eslint-plugin and https://github.com/tleunen/eslint-import-resolver-babel-module).

brandonfancher commented:
Driving/Walking selector is out of alignment and width has been lost, at least on iPhone 6s simulator.

**Now:** 
![simulator screen shot jul 25 2017 11 33 07 am](https://user-images.githubusercontent.com/7911424/28581315-e9711808-712f-11e7-92a8-b7dbe04ac2a4.png)

**Before:** 
![img_0551](https://user-images.githubusercontent.com/7911424/28581406-1bbcaf0c-7130-11e7-9c5b-04da002b3eb7.PNG)

brandonfancher commented:
Black line dividing two sides of station schedules is gone. It actually looks like this bug may have been introduced in PR 1.

**Now:**
![simulator screen shot jul 25 2017 11 35 14 am](https://user-images.githubusercontent.com/7911424/28581512-60da98d8-7130-11e7-9b37-fad7420adaf2.png)

**Before:**
![img_0553](https://user-images.githubusercontent.com/7911424/28581546-7e27012e-7130-11e7-97ad-c59b9cf57fb0.PNG)

mdwagner commented:
Hmm... I'll look into these issues.

brandonfancher commented:
Now that we have Jest tests, might also be a good time to introduce `https://www.npmjs.com/package/eslint-plugin-jest`. That should prevent eslint from yelling at us for `describe`, `it`, and `expect` keywords.

brandonfancher commented:
@mdwagner. Wow, seriously. Good work here. The improvements you're making are really great! Thank you.

After the changes I've outlined are made, I'll review it all once more in more detail and on various iPhone models. Then I'll approve and merge into `development`.

mdwagner commented:
@brandonfancher are we looking to keep eslint? I don't have a problem with using it, I just haven't used it before and currently haven't been interested in using it. If it's something we want to keep, I'll look into improving it, but it hasn't been one of my goals.

brandonfancher commented:
@mdwagner: Yes, we should keep it. It helps us ensure a consistent code style moving forward, and informs new developers working on the project what that style should be. And I've deliberately chosen and further modified AirBNB's ESLint code style.

That being said, if you have other suggestions, I'm all ears. (I'm not a fan of StandardJS though.)

mdwagner commented:
@brandonfancher So I got ESLint working and re-configured for the new dependencies. I also fixed those visual bugs (The alignment & missing vertical line), which pointed out a problem when I converted all styles to Styled-Components, I didn't understand how flex worked in general but I do now. I did try running the app on a couple of different iPhones and didn't see any immediate issues, but feel free to continue testing it on your own. Otherwise, let me know what else I can do for this PR. Thanks!

brandonfancher commented:
Thanks, @mdwagner. I'll review soon and get this merged.

brandonfancher commented:
This was great, @mdwagner. I'm sorry it took me almost a month to merge.

What are your thoughts as to when we should merge this into production and cut a new release to the App Store? I want to get your name and face up on our site as a contributor too soon.
